src/Rule/Nested.php

```diff
@@ -36,9 +37,86 @@
 use function sprintf;
 
 /**
- * Can be used for validation of nested structures.
+ * Used to define rules for validation of nested structures:
+ *
+ * - For one-to-one relation, using `Nested` rule is enough.
+ * - One-to-many and many-to-many relations require pairing with {@see Each} rule.
+ *
+ * An example with blog post:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'title' => [new HasLength(max: 255)],
+ *     // One-to-one relation
+ *     'author' => new Nested([
+ *         'name' => [new HasLength(min: 1)],
+ *     ]),
+ *     // One-to-many relation
+ *     'files' => new Each([
+ *          new Nested([
+ *             'url' => [new Url()],
+ *         ]),
+ *     ]),
+ * ]);
+ * ```
+ *
+ * There is an alternative way to write this using dot notation and shortcuts:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'title' => [new HasLength(max: 255)],
+ *     'author.name' => [new HasLength(min: 1)],
+ *     'files.*.url' => [new Url()],
+ * ]);
+ * ```
+ *
+ * Note that the maximum nesting level is 2, a deeper one requires wrapping with another `Nested` instance or using
+ * short syntax shown above. This will work:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'author' => [
+ *         'name' => [new HasLength(min: 1)],
+ *     ],
+ * ]);
+ * ```
+ *
+ * But this will not:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'author' => [
+ *         'name' => [
+ *             'surname' => [new HasLength(min: 1)],
+ *         ],
+ *     ],
+ * ]);
+ * ```
+ *
+ * Also it's possible to omit arrays for single rules:
+ *
+ *  * ```php
+ * $rules = [
+ *     new Nested([
+ *         'author' => new Nested([
+ *             'name' => new HasLength(min: 1),
+ *         ]),
+ *     ]),
+ * ];
+ * ```
+ *
+ * For more examples please refer to the guide.
+ *
+ * It's also possible to use DTO objects with PHP attributes, see {@see ObjectDataSet} documentation and guide for
+ * details.
+ *
+ * Supports propagation of options (see {@see PropagateOptionsHelper::propagate()} for supported options and
```

Comment:
This sentence doesn't look alright. Is it possible to make it better?

Reply:
Sorry, but could you tell what exactly is wrong in your opinion?